### ISSUE TITLE
no-unused-vars should detect usage in JSX tag

### ISSUE THREAD

**Issue by marians** [eslint/eslint#2805]
The `no-unused-vars` error is incorrectly triggered when a variable is only used in form of a JSX tag.

Example:

```javascript
import React from 'react';
import OtherComponent from './othercomponent';

class MyComponent extends React.Component {
  render() {
    return <OtherComponent />;
  }
}
```

This will generate the error

```
OtherComponent is defined but never used
```

even with the jsx feature activated. Or am I missing anything? Here is my `.eslintrc`:

```json
{
  "ecmaFeatures": {
    "jsx": true,
    "modules": true,
    "classes": true
  },
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "rules": {
    "no-underscore-dangle": 0,
    "quotes": [1, "single"]
  }
}
```

**Comment by hzoo** [comment]
It's recommended you check out https://github.com/yannickcr/eslint-plugin-react for that

**Comment by lo1tuma** [comment]
This works as intended. We don’t support react semantics. You have to use `eslint-plugin-react` for this.

**Comment by marians** [comment]
Thanks for the hint! `eslint-plugin-react` seems to solve this problem.

**Comment by jacobp100** [comment]
If jsx is available as part of the `ecmaFeatures` config, it should really be fixed in the core. This really isn't the expected behaviour, so could this be reopened?

**Comment by ilyavolodin** [comment]
@jacobp100 React is not part of jsx specification. There are other frameworks that use jsx and they do not require this. We do not support React naively, that's why this functionality was moved out into a plugin.
